### PR TITLE
🐛 Implement PackageSettingsPersister.TryFetch

### PR DESCRIPTION
**Pull request by sharwell**
`PackageSettingsPersister.InitializeAsync` is not guaranteed to complete before the constructor returns, so it's possible for `TryFetch` to be called. This method is now implemented to provide a best-known value prior to initialization completing.

### CODE REVIEW

**Review by davidwengier — APPROVED** [review]

**Comment by davidwengier** [comment]
I cherry-picked this and validated that this fixes the break in the new document formatting.

**Review by dibarbet — APPROVED** [review]

File: src/VisualStudio/Core/Def/Implementation/Options/PackageSettingsPersister.cs [hunk 1]
```diff
@@ -56,8 +55,22 @@ private void OnAnalysisScopeChanged(object? sender, EventArgs e)
 
         public bool TryFetch(OptionKey optionKey, out object? value)
         {
-            // This option is refreshed via the constructor to avoid UI dependencies when retrieving option values
-            Contract.ThrowIfTrue(Equals(optionKey.Option, SolutionCrawlerOptions.SolutionBackgroundAnalysisScopeOption));
+            // This option is refreshed via the constructor to avoid UI dependencies when retrieving option values. If
+            // we happen to reach this point before the value is available, try to obtain it without blocking, and
+            // otherwise fall back to the default.
```
Comment (CyrusNajmabadi) [inline src/VisualStudio/Core/Def/Implementation/Options/PackageSettingsPersister.cs:60]: this is worrying to me as i really don't know what implications this has.  can we end up just using an incorrect value that doesn't represent what the system wants?

**Comment by JoeRobich** [comment]
Merging to fix blocking bug.

File: src/VisualStudio/Core/Def/Implementation/Options/PackageSettingsPersister.cs (hunk 1 above)
Comment (sharwell) [reply src/VisualStudio/Core/Def/Implementation/Options/PackageSettingsPersister.cs:60]: We can end up starting with the wrong value, but it doesn't really concern me for several reasons:
1. The trigger for starting with the wrong state involves opening a document, and no matter what option is supposed to be used we know the open document is going to be analyzed.
2. The fallback default is the default state for the IDE as a whole. This is a rarely-changed option, so for most users we start in the correct state no matter what.
3. For the majority of cases where the current solution differs from the default state, the solution state includes _more_ analysis content than the default, so even if the value starts out wrong the IDE will not reach a transient state on the way to steady-state where more diagnostics are shown than desired.

**Pull request merged**